### PR TITLE
chore: added helm chart

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: helm/templates/ingestion-pv.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  name: {{ .Values.local.storage.ingestionSourcePvc.name }}
+  labels:
+    app: "ingestion"
+    component: "ingestion"
+    environment: "development"
+spec:
+  storageClassName: host-path
+  capacity:
+    storage:  15Gi
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:13]: should be configuration

File: helm/templates/ingestion-pv.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ .Values.local.storage.ingestionSourcePvc.name }}
+  labels:
+    app: "ingestion"
+    component: "ingestion"
+    environment: "development"
+spec:
+  storageClassName: host-path
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:11]: should be configuration

File: helm/templates/ingestion-pv.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.local.storage.internalPvc.create -}}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ .Values.local.storage.ingestionSourcePvc.name }}
+  labels:
+    app: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:7]: should be configuration

File: helm/templates/ingestion-pv.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    component: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:8]: should be configuration

File: helm/templates/ingestion-pv.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:9]: should be configuration

File: helm/templates/ingestion-pv.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  storageClassName: host-path
+  capacity:
+    storage:  15Gi
+  accessModes:
+    - ReadWriteOnce
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pv.yaml:15]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.local.storage.ingestionSourcePvc.create -}}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.rasterCommon.storage.fs.ingestionSourcePvc.name }}
+  labels:
+    app: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:7]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    component: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:8]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:9]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+spec:
+  selector: 
+    matchLabels: 
+      app: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:13]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+      component: "ingestion"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:14]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+      environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:15]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+  storageClassName: host-path
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:16]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+  accessModes:
+    - ReadWriteOnce
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:18]: should be configuration

File: helm/templates/ingestion-pvc.yaml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  resources:
+    requests:
+      storage: 4Gi
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/ingestion-pvc.yaml:21]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 16]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.local.storage.internalPvc.create -}}
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ .Values.local.storage.internalPv.name }}
+  labels:
+    app: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:7]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+    component: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:8]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+    environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:9]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  storageClassName: host-path
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:11]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+  capacity:
+    storage:  15Gi
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:13]: should be configuration

File: helm/templates/internal-pv.yaml [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+  accessModes:
+    - ReadWriteOnce
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pv.yaml:15]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 22]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.local.storage.internalPvc.create -}}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.rasterCommon.storage.fs.internalPvc.name }}
+  labels:
+    app: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:7]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+    component: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:8]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+    environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:9]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+spec:
+  selector: 
+    matchLabels: 
+      app: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:13]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+      component: "internal"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:14]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+      environment: "development"
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:15]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+  storageClassName: host-path
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:16]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 29]
```diff
@@ -0,0 +1,22 @@
...
+  accessModes:
+    - ReadWriteOnce
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:18]: should be configuration

File: helm/templates/internal-pvc.yaml [hunk 30]
```diff
@@ -0,0 +1,22 @@
...
+  resources:
+    requests:
+      storage: 4Gi
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/internal-pvc.yaml:21]: should be configuration

File: helm/templates/service.yaml [hunk 31]
```diff
@@ -0,0 +1,32 @@
+{{- $releaseName := .Release.Name -}}
```
Comment (cl-shaharshavit-rnd) [inline helm/templates/service.yaml:1]: remove service

File: helm/values.yaml [hunk 32]
```diff
@@ -0,0 +1,80 @@
...
+      internalPvc:
+        name: "internal-pvc"
+        tilesSubPath: tiles
+    s3:
+      endpointUrl: localhost:9000"
+      tilesBucket: ""
+      secretName: "s3-secret"
+
+env:
+  port: 80
```
Comment (cl-shaharshavit-rnd) [inline helm/values.yaml:56]: remove

File: helm/values.yaml [hunk 33]
```diff
@@ -0,0 +1,80 @@
...
+  targetPort: 8080
```
Comment (cl-shaharshavit-rnd) [inline helm/values.yaml:57]: make const

File: helm/values.yaml [hunk 34]
```diff
@@ -0,0 +1,80 @@
...
+  protocol: TCP
```
Comment (cl-shaharshavit-rnd) [inline helm/values.yaml:58]: make const

File: helm/values.yaml [hunk 35]
```diff
@@ -0,0 +1,80 @@
...
+  jobType: ""
+  pollingTime: "1000"
+  logging:
+    logLevel:
+      default: "Information"
+      aspNetCore: "Warning"
+  allowedHosts: "*"
+
+resources:
+  enabled: false
```
Comment (cl-shaharshavit-rnd) [inline helm/values.yaml:72]: should be true by default

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: helm/templates/configmap.yaml [hunk 36]
```diff
@@ -0,0 +1,19 @@
...
+  GENERAL__validate: {{ .Values.env.general.validate | quote }}
+  GENERAL__outputDir: {{ .Values.env.general.outputDir | quote }}
+  TASK__types__taskType: {{ .Values.env.taskType | quote }}
+  TASK__types__jobType: {{ .Values.env.jobType | quote }}
+  TASK__pollingTime: {{ .Values.env.pollingTime | quote }}
+  S3__url: {{ .Values.rasterCommon.storage.s3.endpointUrl | quote }}
+  S3__bucket: {{ .Values.rasterCommon.storage.s3.tilesBucket | quote }}
+  Logging__LogLevel__Default: {{ .Values.env.logging.logLevel.default | quote }}
+  Logging__Microsoft.AspNetCore: {{ .Values.env.logging.logLevel.aspNetCore | quote }}
+  AllowedHosts: {{ .Values.env.allowedHosts | quote }}
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/templates/configmap.yaml:18]: fixed

File: helm/templates/ingestion-pv.yaml (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:13]: fixed

File: helm/templates/ingestion-pv.yaml (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:11]: fixed

File: helm/values.yaml (hunk 35 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:72]: fixed

File: helm/templates/service.yaml (hunk 31 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/service.yaml:1]: removed

File: helm/templates/ingestion-pv.yaml (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:7]: removed

File: helm/templates/ingestion-pv.yaml (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:8]: removed

File: helm/templates/ingestion-pv.yaml (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:9]: removed

File: helm/templates/ingestion-pvc.yaml (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:7]: fixed

File: helm/templates/ingestion-pv.yaml (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pv.yaml:15]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 8 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:8]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 9 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:9]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 11 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:14]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 10 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:13]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 12 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:15]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 13 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:16]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 14 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:18]: fixed

File: helm/templates/ingestion-pvc.yaml (hunk 15 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/ingestion-pvc.yaml:21]: fixed

File: helm/templates/internal-pv.yaml (hunk 17 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:8]: fixed

File: helm/templates/internal-pv.yaml (hunk 18 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:9]: removed

File: helm/templates/internal-pv.yaml (hunk 19 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:11]: fixwed

File: helm/templates/internal-pv.yaml (hunk 20 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:13]: fixed

File: helm/templates/internal-pvc.yaml (hunk 22 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:7]: fixed

File: helm/values.yaml (hunk 32 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:56]: removed

File: helm/values.yaml (hunk 33 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:57]: fixed

File: helm/values.yaml (hunk 34 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml:58]: fixed

File: helm/templates/internal-pv.yaml (hunk 21 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:15]: fixed

File: helm/templates/internal-pv.yaml (hunk 16 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pv.yaml:7]: removed

File: helm/templates/internal-pvc.yaml (hunk 23 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:8]: fixed

File: helm/templates/internal-pvc.yaml (hunk 24 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:9]: fixed

File: helm/templates/internal-pvc.yaml (hunk 25 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:13]: fixed

File: helm/templates/internal-pvc.yaml (hunk 27 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:15]: fixed

File: helm/templates/internal-pvc.yaml (hunk 26 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:14]: fixed

File: helm/templates/internal-pvc.yaml (hunk 28 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:16]: fixed

File: helm/templates/internal-pvc.yaml (hunk 29 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:18]: fixed

File: helm/templates/internal-pvc.yaml (hunk 30 above)
Comment (CL-SHLOMIKONCHA) [reply helm/templates/internal-pvc.yaml:21]: fixed

File: helm/templates/configmap.yaml [hunk 37]
```diff
@@ -0,0 +1,19 @@
+{{- $chartName := include "gpkg-merger.name" . -}}
+
+{{- if .Values.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-{{ $chartName }}-configmap
+data:
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/templates/configmap.yaml:8]: fixed

File: helm/templates/deployment.yaml [hunk 38]
```diff
@@ -0,0 +1,86 @@
...
+        persistentVolumeClaim:
+          claimName: {{ .Values.rasterCommon.storage.fs.internalPvc.name }}
+    #   nodeSelector:
+    #     knownNode: eastus-1-node-1
+      containers:
+        - name: {{ $releaseName }}-{{ $chartName }}-deployment
+          {{- with .Values.image }}
+          image: {{ $cloudProviderDockerRegistryUrl }}{{ .repository }}:{{ .tag }}
+          {{- end }}
+          env:
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply helm/templates/deployment.yaml:56]: fixed

File: helm/values.yaml [hunk 39]
```diff
@@ -0,0 +1,85 @@
...
+  cloudProvider: {}
+  tracing: {}
+  metrics: {}
+  environment: {}
+
+enabled: true
+environment: development
+replicaCount: 1
+initialDelaySeconds: 60
+nodePort: 30022 #for minikube deployment only
```
Comment (cl-shaharshavit-rnd) [inline helm/values.yaml:11]: remove

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: helm/values.yaml (hunk 39 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml]: removed

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: helm/values.yaml [hunk 40]
```diff
@@ -0,0 +1,92 @@
...
+      {
+        taskType: "tilesMerging",
+        jobType: "Ingestion_Update"
+      },
+      {
+        taskType: "tilesMerging",
+        jobType: "Ingestion_New"
+      }
+    ]
+  pollingTime: "1000"
```
Comment (shimoncohen) [inline helm/values.yaml:77]: Remove.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

**Review by shimoncohen — APPROVED** [review]

File: helm/values.yaml (hunk 40 above)
Comment (CL-SHLOMIKONCHA) [reply helm/values.yaml]: removed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**